src/hera/workflows/_mixins.py

```diff
@@ -527,6 +501,87 @@ def __call__(self, *args, **kwargs) -> Optional[SubNodeMixin]:
 
         raise InvalidTemplateCall("Container is not under a Steps, Parallel, or DAG context")
 
+    def _get_arguments(self, **kwargs) -> list:
```

Comment:
Should use the typing module types for now (same in `isinstance` below and for _get_parameter_names and _get_artifact_names args)
```suggestion
    def _get_arguments(self, **kwargs) -> List:
```